google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/StatusUtil.java

```diff
@@ -17,21 +17,31 @@
 package com.google.cloud.pubsub.spi.v1;
 
 import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
 
 /** Utilities for handling gRPC {@link Status}. */
 final class StatusUtil {
   private StatusUtil() {
-    // Static class, not instatiable.
+    // Static class, not instantiable.
   }
 
-  public static boolean isRetryable(Status status) {
-    switch (status.getCode()) {
+  public static boolean isRetryable(Throwable error) {
+    if (!(error instanceof StatusRuntimeException)) {
+      return true;
+    }
+    StatusRuntimeException statusRuntimeException = (StatusRuntimeException) error;
+    switch (statusRuntimeException.getStatus().getCode()) {
       case DEADLINE_EXCEEDED:
       case INTERNAL:
       case CANCELLED:
       case RESOURCE_EXHAUSTED:
-      case UNAVAILABLE:
         return true;
+      case UNAVAILABLE:
+        if (statusRuntimeException.getMessage().contains("Server shutdownNow invoked")) {
```

Comment:
I'm not sure what you mean by "special status", but when `UNAVAILABLE` is thrown for this reason, it can't be retried. I'm not sure how stable the message is. This particular change isn't strictly necessary because the `!isAlive()` code path will exit `onFailure` before this logic is called. If you're concerned, I can remove it.